vm/src/builtins/str.rs

```diff
@@ -509,8 +509,13 @@ impl PyStr {
     #[pymethod(name = "__rmul__")]
     #[pymethod(magic)]
     fn mul(zelf: PyRef<Self>, value: isize, vm: &VirtualMachine) -> PyResult<PyRef<Self>> {
-        if value == 1 && zelf.class().is(&vm.ctx.types.str_type) {
-            // Special case: when some `str` is multiplied by `1`,
+        if value == 0 && zelf.class().is(&vm.ctx.types.str_type) {
+            // Special case: when some `str` is multiplied by `0`,
+            // returns an empty `str`.
+            return Ok(PyStr::from("").into_ref(vm));
```

Comment:
The optimization is expected to reuse the shared PyStr object, instead of creating new. (PyStr::from)

The expected test to pass is `('' * 3) is ''` and `('a' * 0) is ''`

You can add how we handle `empty_tuple` or `empty_frozenset`. Search for the name in the source code, then you will find the way to make `empty_str`.

Reply:
Thank you, i got it.

Reply:
@youknowone Try to fix it again, please take a look, thank you.